lib/karafka/serialization/json/deserializer.rb

```diff
@@ -7,12 +7,17 @@ module Serialization
     module Json
       # Default Karafka Json deserializer for loading JSON data
       class Deserializer
-        # @param content [String] content based on which we want to get our hash
+        # @param params [Karafka::Params::Params] Full params object that we want to deserialize
         # @return [Hash] hash with deserialized JSON data
         # @example
-        #   Deserializer.call("{\"a\":1}") #=> { 'a' => 1 }
-        def call(content)
-          ::MultiJson.load(content)
+        #   params = {
+        #     'payload' => "{\"a\":1}",
+        #     'topic' => 'my-topic',
+        #     'headers' => { 'message_type' => :test }
+        #   }
+        #   Deserializer.call(params) #=> { 'a' => 1 }
+        def call(params)
+          ::MultiJson.load(params['payload'])
```

Comment:
`params.payload`

Reply:
@nijikon can we benchmark which one is faster? I have a feeling, that `params[]` will be slightly faster.

Reply:
We can if it will be faster then I would opt for removing methods.

Reply:
Added a card for this. Let's leave it as is now.